expertsystem/amplitude/canonical_decay.py

```diff
@@ -4,7 +4,7 @@
 
 from expertsystem.data import Spin
 from expertsystem.solving.topology import StateTransitionGraph
-from expertsystem.types import InteractionProperties, ParticleWithSpin
+from expertsystem.solving.types import InteractionProperties, ParticleWithSpin
```

Comment:
Ok good, so my first comment is already addressed. I personally don't like the name types very much. Maybe rename it into `quantum_number` or something more meaningful?